influxdb_iox/src/clap_blocks/catalog_dsn.rs

```diff
@@ -1,24 +1,46 @@
-use std::sync::Arc;
-
 use iox_catalog::{
+    create_or_get_default_records,
     interface::{Catalog, Error},
+    mem::MemCatalog,
     postgres::PostgresCatalog,
 };
+use std::sync::Arc;
 
 /// CLI config for catalog DSN.
 #[derive(Debug, Clone, clap::Parser)]
 pub struct CatalogDsnConfig {
-    /// Postgres connection string
+    /// The type of catalog to use.
+    #[clap(arg_enum, long = "--catalog", env = "INFLUXDB_IOX_CATALOG_TYPE")]
```

Comment:
```suggestion
    #[clap(arg_enum, long = "--catalog", env = "INFLUXDB_IOX_CATALOG_TYPE", default_value = "postgres")]
```

I think we should somehow not require toooo many args to be passed by default and "in-mem" doesn't really make sense for any sensible server.